stl/STL.natvis

```diff
@@ -0,0 +1,1596 @@
+      <DisplayString Condition="_MyPromise._State._Assoc_state != nullptr &amp;&amp; (*_MyPromise._State._Assoc_state)._Ready == 0">pending</DisplayString>
+      <DisplayString Condition="_MyPromise._State._Assoc_state != nullptr &amp;&amp; (*_MyPromise._State._Assoc_state)._Ready == 1">has_result</DisplayString>
+      <Expand>
+          <ExpandedItem Condition="_MyPromise._State._Assoc_state != nullptr &amp;&amp; (*_MyPromise._State._Assoc_state)._Ready == 1">*_MyPromise._State._Assoc_state</ExpandedItem>
+          <Item Name="[future_retrieved]" Condition="_MyPromise._State._Assoc_state != nullptr">_MyPromise._Future_retrieved</Item>
+      </Expand>
+  </Type>
+
+  <!-- VC 2015 -->
+  <Type Name="std::future&lt;*&gt;">
+      <AlternativeType Name="std::shared_future&lt;*&gt;"/>
+      <DisplayString Condition="_Assoc_state == nullptr">empty</DisplayString>
+      <DisplayString Condition="_Assoc_state != nullptr &amp;&amp; (*_Assoc_state)._Ready == 0">pending</DisplayString>
+      <DisplayString Condition="_Assoc_state != nullptr &amp;&amp; (*_Assoc_state)._Ready == 1">has_result</DisplayString>
+      <Expand>
+          <ExpandedItem Condition="_Assoc_state != nullptr &amp;&amp; (*_Assoc_state)._Ready == 1">*_Assoc_state</ExpandedItem>
+      </Expand>
+  </Type>
+
+
+  <Type Name="std::pointer_to_unary_function&lt;*&gt;">
+      <AlternativeType Name="std::pointer_to_binary_function&lt;*&gt;" />
+      <DisplayString>ptr_fun({_Pfun})</DisplayString>
+      <Expand/>
+  </Type>
+
+  <Type Name="std::mem_fun_t&lt;*&gt;">
+      <AlternativeType Name="std::mem_fun1_t&lt;*&gt;" />
+      <AlternativeType Name="std::const_mem_fun_t&lt;*&gt;" />
+      <AlternativeType Name="std::const_mem_fun1_t&lt;*&gt;" />
+      <DisplayString>mem_fun({_Pmemfun})</DisplayString>
+      <Expand/>
+  </Type>
+
+  <Type Name="std::mem_fun_ref_t&lt;*&gt;">
+      <AlternativeType Name="std::mem_fun1_ref_t&lt;*&gt;" />
+      <AlternativeType Name="std::const_mem_fun_ref_t&lt;*&gt;" />
+      <AlternativeType Name="std::const_mem_fun1_ref_t&lt;*&gt;" />
+      <DisplayString>mem_fun_ref({_Pmemfun})</DisplayString>
+      <Expand/>
+  </Type>
+
+  <Type Name="std::binder1st&lt;*&gt;">
+      <DisplayString>bind1st({op}, {value})</DisplayString>
+      <Expand>
+          <Item Name="op">op</Item>
+          <Item Name="value">value</Item>
+      </Expand>
+  </Type>
+
+  <Type Name="std::binder2nd&lt;*&gt;">
+      <DisplayString>bind2nd({op}, {value})</DisplayString>
+      <Expand>
+          <Item Name="op">op</Item>
+          <Item Name="value">value</Item>
+      </Expand>
+  </Type>
+
+  <Type Name="std::auto_ptr&lt;*&gt;">
+      <DisplayString Condition="_Myptr == 0">empty</DisplayString>
+      <DisplayString Condition="_Myptr != 0">auto_ptr {*_Myptr}</DisplayString>
+      <Expand>
+          <Item Condition="_Myptr != 0" Name="[ptr]">_Myptr</Item>
+      </Expand>
+  </Type>
+
+
+  <!--Visualizer for coroutines-->

```

Comment:
MSVC-PR-307511 is making changes here. Should we wait for that PR to land and merge those changes here, or flow those changes in a separate PR later?

Reply:
Fun indeed, I agree we should wait for that changes to go in

Reply:
I see the coroutine changes here but I think we're still missing [VS-PR-308324](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/308324).

Reply:
> I see the coroutine changes here but I think we're still missing [VS-PR-308324](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/308324).

You are blind; I see both the `string_view` and `vector` changes here from that PR.